### ISSUE TITLE
Skipped tests in CTest register as Error

### ISSUE THREAD

**Issue by fweik** [espressomd/espresso#528]
Seems to be an upstream issue:
https://cmake.org/Bug/bug_relationship_graph.php?bug_id=15431

**Comment by rbardak** [comment]
Heres the discussion for the skip return code feature request of cmake
https://cmake.org/Bug/view.php?id=8466

looking at the diff from the commit that added the skip return value feature
https://cmake.org/gitweb?p=cmake.git;a=commitdiff;h=3a0d164bb2fa03966cdd7a17c0ce63b4d54f05b7
You can see that a skipped test is handled as a "not run" test. The "not run" status was initially for tests which could not find the required files, which is supposed to be a failed test.

There are also some discussions about it in the mailing list
https://cmake.org/pipermail/cmake/2012-June/050661.html

TL;DR: it's not a bug, it's a feature

One way to avoid a failed test is to write a wrapper script for the tests which intercepts the skip return code and reports it as passed, then again that would be misinformation.

The only other alternative would ne not to rely on ctest and do it old school.

**Comment by fweik** [comment]
The code of CTest (Source/CTest/cmCTestRunTest.cxx) seems to suggest that skipped tests should pass. For me this looks like a bug.

**Comment by rbardak** [comment]
in the current master of the repository https://cmake.org/cmake.git
for Source/CTest/cmCTestRunTest.cxx in line 214 you can see that the test result is set to "not run" in the case of a return code skip. 
A little later, in line 309 this activates an if-statement which sets the boolean pass to false. This looks pretty intentional to me.

**Comment by fweik** [comment]
After a review I tend to let skipped test exit as success. I believe the information loss is minor and its not worthwhile to put effort into this.
